src/Components/Artwork/Details/__tests__/PrimaryLabelLine.jest.tsx

```diff
@@ -0,0 +1,209 @@
+
+jest.unmock("react-relay")
+
+jest.mock("System/Hooks/useSystemContext")
+jest.mock("Components/ArtworkGrid/ArtworkGridContext")
+jest.mock("Utils/Hooks/useDidMount")
+jest.mock("Utils/Hooks/useIntersectionObserver")
+
+describe("PrimaryLabelLine", () => {
+  let mockuseDidMount = useDidMount as jest.Mock
+  let mockUseIntersectionObserver = useIntersectionObserver as jest.Mock
+  let mockUseSystemContext = useSystemContext as jest.Mock
+  let mockUseArtworkGridContext = useArtworkGridContext as jest.Mock
+  let mockEnvironment
+
+  beforeEach(() => {
+    mockuseDidMount.mockImplementation(() => true)
+
+    let called = false
+    mockUseIntersectionObserver.mockImplementation(({ onIntersection }) => {
+      if (!called) {
+        onIntersection()
+        called = true
+      }
+      return { ref: createRef() }
+    })
+
+    mockEnvironment = createMockEnvironment()
+    mockUseSystemContext.mockImplementation(() => {
+      return {
+        relayEnvironment: mockEnvironment,
+      }
+    })
+    mockUseArtworkGridContext.mockImplementation(() => {
+      return {}
+    })
+  })
+
+  describe("before fetching partner offer", () => {
+    it("shows increased interest when it's passed in as the label", () => {
+      render(
+        <PrimaryLabelLineQueryRenderer
+          id="opaque-internal-id"
+          label="INCREASED_INTEREST"
+        />
+      )
+
+      expect(screen.getByText("Increased Interest")).toBeInTheDocument()
+    })
+
+    it("shows curators' pick when it's passed in as the label", () => {
+      render(
+        <PrimaryLabelLineQueryRenderer
+          id="opaque-internal-id"
+          label="CURATORS_PICK"
+        />
+      )
+
+      expect(screen.getByText("Curators’ Pick")).toBeInTheDocument()
+    })
+
+    it("hides the increased interest label when under hide signals context", () => {
+      mockUseArtworkGridContext.mockImplementation(() => {
+        return { hideSignals: true }
+      })
+
+      const { container } = render(
+        <PrimaryLabelLineQueryRenderer
+          id="opaque-internal-id"
+          label="INCREASED_INTEREST"
+        />
+      )
+
+      expect(container).toBeEmptyDOMElement()
+    })
+
+    it("hides the curators' pick label when under hide signals context", () => {
+      mockUseArtworkGridContext.mockImplementation(() => {
+        return { hideSignals: true }
+      })
+
+      const { container } = render(
+        <PrimaryLabelLineQueryRenderer
+          id="opaque-internal-id"
+          label="CURATORS_PICK"
+        />
+      )
+
+      expect(container).toBeEmptyDOMElement()
+    })
+  })
+
+  describe("after fetching null partner offer", () => {
+    const renderPrimaryLabelLine = label => {
+      const { container } = render(
+        <PrimaryLabelLineQueryRenderer id="opaque-internal-id" label={label} />
+      )
+
+      act(() => {
+        mockEnvironment.mock.resolveMostRecentOperation(operation =>
+          MockPayloadGenerator.generate(operation, {
```

Comment:
For example, this stuff can just be provided as props to the fragment component in the test helper setup.